### PR TITLE
Exit with non-zero number if there is some error with prepare.sh

### PR DESCRIPTION
**Pull request by ZiyueHuang**

### CODE REVIEW

**Comment by ZiyueHuang** [comment]
Please refer to https://github.com/intel-hadoop/HiBench/issues/457

File: bin/run_all.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ for benchmark in `cat $root_dir/conf/benchmarks.lst`; do
     if [ $? -ne 0 ]
     then
 	echo "ERROR: ${benchmark} prepare failed!"
-        continue
+        exit $?
```
Comment (carsonwang) [inline bin/run_all.sh]: `$?` here is not the result of prepare.sh, right? We may need to save the result first like `result=$`?
Can you please test it by failing one prepare.sh and see Travis's result?

Comment (ZiyueHuang) [inline bin/run_all.sh]: @carsonwang Sorry for this mistake, you are right. I'll test as you say right now.

**Comment by ZiyueHuang** [comment]
As you can see in travis log, if prepare.sh fails then build failing.
```
Exec script: /home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 21: /home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/../../../../..//bin/functions/load-bench_config.sh: No such file or directory
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 23: enter_bench: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 24: show_bannar: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 26: rmr_hdfs: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 27: timestamp: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 37: run_hadoop_job: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 38: timestamp: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 40: show_bannar: command not found
/home/travis/build/intel-hadoop/HiBench/bin/workloads/ml/bayes/prepare/prepare.sh: line 41: leave_bench: command not found
ERROR: ml/bayes prepare failed!
The command "sudo -E ./bin/run_all.sh" exited with 127.
cache.2
store build cache
0.00s
78.03schange detected (content changed, file is created, or file is deleted):
/home/travis/.m2/repository/org/apache/maven/plugins/maven-shade-plugin/resolver-status.properties
changes detected, packing new archive
.
.
.

uploading archive
Done. Your build exited with 1.
```

**Comment by carsonwang** [comment]
LGTM. Thanks for the fix!

**Pull request merged**